tests/core/actionrun_test.py

```diff
@@ -738,24 +740,37 @@ def test_handler_unhandled(self):
         ) is None
         assert self.action_run.is_scheduled
 
-    def test_recover_action_run_no_action_runner(self):
+    def test_recover_no_action_runner(self):
         # Default setup has no action runner
         assert not self.action_run.recover()
 
-    def test_recover_action_run_incorrect_state(self):
-        # Should return falsy if not UNKNOWN.
-        self.action_run.action_runner = SubprocessActionRunnerFactory(
+
+class TestSSHActionRunRecover:
+    @pytest.fixture(autouse=True)
+    def setup_action_run(self):
+        self.output_path = filehandler.OutputPath(tempfile.mkdtemp())
```

Comment:
can you wrap this in another decorator so we know for sure that the tmp dir will get cleaned up?

Reply:
I pulled output_path out into a separate fixture - let me know if that's not what you were thinking.

Reply:
merging this by popular demand, but I can always update it later.